OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignalSyncServiceUtils.java

```diff
@@ -142,20 +141,21 @@ private static boolean hasBootPermission(Context context) {
    private static boolean isJobIdRunning(Context context) {
       final JobScheduler jobScheduler = (JobScheduler) context.getSystemService(Context.JOB_SCHEDULER_SERVICE);
       for (JobInfo jobInfo : jobScheduler.getAllPendingJobs()) {
-         if (jobInfo.getId() == OneSignalSyncServiceUtils.SYNC_TASK_ID) {
+         if (jobInfo.getId() == OneSignalSyncServiceUtils.SYNC_TASK_ID && syncBgThread != null && syncBgThread.isAlive()) {
             return true;
          }
       }
       return false;
    }
 
    @RequiresApi(21)
-   private static boolean scheduleSyncServiceAsJob(Context context, long delayMs) {
+   private static void scheduleSyncServiceAsJob(Context context, long delayMs) {
       OneSignal.Log(OneSignal.LOG_LEVEL.VERBOSE, "scheduleSyncServiceAsJob:atTime: " + delayMs);
 
       if (isJobIdRunning(context)) {
          OneSignal.Log(OneSignal.LOG_LEVEL.VERBOSE, "scheduleSyncServiceAsJob Scheduler already running!");
-         return false;
+         needsJobReschedule = true;
```

Comment:
Could use a comment for this if block of why we can't schedule another job now, and that we will once the current job is finished.